docs/how-to/hip_porting_driver_api.rst

```diff
@@ -0,0 +1,361 @@
+
+``cuCtx`` API
+-------------
+
+The Driver API defines "Context" and "Devices" as separate entities.
+Contexts contain a single device, and a device can theoretically have multiple contexts.
+Each context contains a set of streams and events specific to the context.
+Historically contexts also defined a unique address space for the GPU, though this may no longer be the case in Unified Memory platforms, since the CPU and all the devices in the same process share a single unified address space.
+The Context APIs also provide a mechanism to switch between devices, which allowed a single CPU thread to send commands to different GPUs.
+HIP as well as a recent versions of CUDA Runtime provide other mechanisms to accomplish this feat - for example using streams or ``cudaSetDevice``.
+
+The CUDA Runtime API unifies the Context API with the Device API. This simplifies the APIs and has little loss of functionality since each Context can contain a single device, and the benefits of multiple contexts has been replaced with other interfaces.
+HIP provides a context API to facilitate easy porting from existing Driver codes.
+In HIP, the ``Ctx`` functions largely provide an alternate syntax for changing the active device.
+
+Most new applications will prefer to use ``hipSetDevice`` or the stream APIs, therefore HIP has marked ``hipCtx`` APIs as **deprecated**. Support for these APIs may not be available in future releases. For more details on deprecated APIs please refer :doc:`reference/deprecated_api_list`.
+
+HIP Module and ``Ctx`` APIs
+===========================
+
+Rather than present two separate APIs, HIP extends the HIP API with new APIs for Modules and ``Ctx`` control.
+
+``hipModule`` API
+-----------------
+
+Like the CUDA Driver API, the Module API provides additional control over how code is loaded, including options to load code from files or from in-memory pointers.
+NVCC and HIP-Clang target different architectures and use different code object formats: NVCC is ``cubin`` or ``ptx`` files, while the HIP-Clang path is the ``hsaco`` format.
+The external compilers which generate these code objects are responsible for generating and loading the correct code object for each platform.
+Notably, there is not a fat binary format that can contain code for both NVCC and HIP-Clang platforms. The following table summarizes the formats used on each platform:
+
+.. list-table:: Module formats
+   :header-rows: 1
+
+   * - Format
+     - APIs
+     - NVCC
+     - HIP-CLANG
+   * - Code Object
+     - ``hipModuleLoad``, ``hipModuleLoadData``
+     - ``.cubin`` or PTX text
+     - ``.hsaco``
+   * - Fat Binary
+     - ``hipModuleLoadFatBin``
+     - ``.fatbin``
+     - ``.hip_fatbin``
+
+``hipcc`` uses HIP-Clang or NVCC to compile host codes. Both of these may embed code objects into the final executable, and these code objects will be automatically loaded when the application starts.
```

Comment:
"codes" is a very uncommon plural when used for source code
```suggestion
``hipcc`` uses HIP-Clang or NVCC to compile host code. Both of these may embed code objects into the final executable, and these code objects will be automatically loaded when the application starts.
```

Reply:
Fixed.